### ISSUE TITLE
Invalid model on UF formula

### ISSUE THREAD

**Issue by wintered** [Z3Prover/z3#3669]
```
[540] % z3 model_validate=true small.smt2
sat
(error "line 4 column 21: an invalid model was generated")
[541] % 
[541] % cat small.smt2
(declare-fun a (Int) Bool)
(assert (forall ((b Int)) (a (div 2 b))))
(assert (a 1))
(check-sat-using ufbv)
[542] %
```
Z3 gave the following clearly incorrect model
```
(model 
 (define-fun a ((x!0 Int)) Bool
  (= x!0 (div 2 x!0)))
)
```

OS: Ubuntu 18.04
Commit: 0dc5bad

**Comment by NikolajBjorner** [comment]
while this isn't qfbv, the macro manager seems to be faulty independent of the tactic abuse.

**Comment by NikolajBjorner** [comment]
fixed with #3705
